keras/models.py

```diff
@@ -77,15 +77,18 @@ def standardize_weights(y, sample_weight=None, class_weight=None):
     if sample_weight is not None:
         return standardize_y(sample_weight)
     elif isinstance(class_weight, dict):
-        if len(y.shape) > 2:
-            raise Exception('class_weight not supported for 3+ dimensional targets.')
+        if len(y.shape) > 3:
+            raise Exception('class_weight not supported for 4+ dimensional targets.')
+        yshape = y.shape
+        y = np.reshape(y, (-1,yshape[-1])) # for time-distributed data, collapse time and sample
```

Comment:
I take a matrix of shape (samples, timesteps, dimensions), collapse it down to (samples*timesteps, dimensions), and then do the class weighting as before. Now i have something of shape (samples*timesteps, 1) which I then reshape to (samples, timesteps, 1).